sympy/core/function.py

```diff
@@ -707,8 +707,7 @@ def fdiff(self, argindex=1):
             else:
                 return Derivative(self, self.args[argindex - 1], evaluate=False)
         # See issue 4624 and issue 4719 and issue 5600
-        arg_dummy = Dummy('xi_%i' % argindex)
-        arg_dummy.dummy_index = hash(self.args[argindex - 1])
+        arg_dummy = Dummy('xi_%i' % argindex, shash=abs(hash(self.args[argindex - 1])))
```

Comment:
I didn't find a test, but I think I know a situation that uses this.  I will file another issue for a test.